bin/spiced/src/lib.rs

```diff
@@ -42,6 +42,18 @@ pub struct Args {
 pub async fn run(args: Args) -> Result<()> {
     let app = App::new(".").context(UnableToConstructSpiceAppSnafu)?;
 
+    let mut auth = runtime::auth::AuthProviders::default();
+    match auth.parse_from_config() {
+        Ok(()) => {}
+        Err(e) => {
+            tracing::error!(
+                "Unable to parse auth from config, proceeding without auth: {}",
+                e
+            );
+        }
+    }
+    tracing::info!("Spice Auth token: {}", auth.get("spice.ai").get_token());
```

Comment:
Do we plan a scenario when we will want to use several providers in the same time so we would like to have `auth.get("spice.ai").get_token()` (as it is proposed here) vs single auth provider instance so it is better to do `auth = runtime::auth::AuthProviders::get("spice.ai")` and then just 'auth.get_token()'


Is it a universal auth we will use to communicate with different services so need support for multiple providers for the same auth instance?